src/api/accounts.rs

```diff
@@ -58,27 +60,35 @@ pub struct AllAccountsRequest {
 impl Request for SingleAccountRequest {
     type Response = resources::Account;
 
-    fn uri(&self, host: &Url) -> Result<Url> {
-        let path = format!("/accounts/{}", self.account_id);
-        Ok(host.join(&path)?)
+    fn uri(&self, base_url: &Url) -> Result<Url> {
+        let mut base_url = base_url.clone();
+        {
+            let mut segments = base_url.path_segments_mut().unwrap();
+            segments.extend(&[API_PATH, self.account_id.as_str()]);
+        }
+        Ok(base_url)
```

Comment:
I think a simpler fix might have been to just remote the initial `/` from the path? So this code should do it:
```rust
let path = format!("accounts/{}", self.account_id);
Ok(host.join(&path)?)
```
Could you give that a try? If it works I'd prefer that since it keeps the code much simpler.

Reply:
> I think a simpler fix might have been to just remove the initial `/` from the path? So this code should do it:
> 
> ```rust
> let path = format!("accounts/{}", self.account_id);
> Ok(host.join(&path)?)
> ```
> 
> Could you give that a try? If it works I'd prefer that since it keeps the code much simpler.

I did try before but the problem is not the `/` but the `join` itself.
What we do here is extend the path, not join it. The `join` works more like if it was a filesystem path, it removes "directory" names from the path, and it sees everything w/o a `/` as a file. I do not like this wordy solution but this is the only way I managed to make it work with an arbitrary URL. I came to the conclusion that the original code worked only because the provided URL was always a host, and `join` has very specific rules about it. Additionally, using `extend` is safer in terms of `/` handling.

Reply:
Alright, I did some testing on this, and it seems `Url::join` would need a trailing `/` to join the URL like you would expect:
```rust
fn main() {
    use url::Url;
    println!("{}", Url::parse("https://example.com/a/b").unwrap().join("c").unwrap());
    println!("{}", Url::parse("https://example.com/a/b/").unwrap().join("c").unwrap());
}
```
outputs:
```
https://example.com/a/c
https://example.com/a/b/c
```
Would it be an option for you to set the base URL with a trailing `/`? If so I'd rather prefer to keep using `join` since the code is much easier to read that way.